build.psm1

```diff
@@ -2437,21 +2437,24 @@ function Start-CrossGen {
         Generate-CrossGenAssembly -CrossgenPath $crossGenPath -AssemblyPath $assemblyPath
     }
 
-    <#
-     # The latest dotnet.exe from .NET Core 1.1.0 preview packages starts
-     # to check the existence of all TPA assemblies, including those built
-     # from local projects (e.g. powershell assemblies). The TPA assemblies
-     # in '.deps.json' are IL assemblies, so we cannot remove those ILs.
-     #
-    Write-Verbose "PowerShell Ngen assemblies have been generated, deleting ILs..." -Verbose
+    #
+    # With the latest dotnet.exe, the default load context is only able to load TPAs, and TPA
+    # only contains IL assembly names. In order to make the default load context able to load
+    # the NI PS assemblies, we need to replace the IL PS assemblies with the corresponding NI
+    # PS assemblies, but with the same IL assembly names.
+    #
+    Write-Verbose "PowerShell Ngen assemblies have been generated. Deploying ..." -Verbose
     foreach ($assemblyName in $psCoreAssemblyList) {
         # Remove the IL assembly and its symbols.
         $assemblyPath = Join-Path $PublishPath $assemblyName
```

Comment:
It is a minor naming thing, but can you rename $assemblyPath and $symbolsPath to $assemblyFile and $symbolFile? In my mind, a name that ends in "path" implies that it is a path to a directory, not a file.